django/views/generic/base.py

```diff
@@ -30,6 +32,27 @@ def get_context_data(self, **kwargs):
         return kwargs
 
 
+class ViewType(Enum):
+    SYNC = auto()
+    ASYNC = auto()
+
+    @classmethod
+    def of_view(cls, view):
+        handlers = [
+            getattr(view, method)
+            for method in view.http_method_names
+            if (method != "options" and hasattr(view, method))
```

Comment:
What about views with a custom `options()` implementation? :thinking:

Reply:
Nice... 😃

A View with **just** `asyc def options()` wouldn't work unless you also overrode `as_view()` to mark the callback correctly. I'm inclined to just live with that.

Otherwise as long as your `options` matched the other handler methods (either `def` or `async def`) it would be fine. We only iterate all the handlers here to add the sanity check in. Presumably if you're writing a custom `options`, you're also writing a test for it, which would fail if your options method didn't match your other handlers. 

I'm not sure how much code this is worth... 🤔